Fushigi/ui/widgets/UndoWindow.cs

```diff
@@ -23,24 +25,37 @@ public void Render(CourseAreaEditContext context)
                 {
                     context.Redo();
                 }
-                foreach (var op in context.GetUndoStack().Reverse())
+                if (ImGui.Selectable($"{IconUtil.ICON_FILE_DOWNLOAD}"+"Course Loaded", !context.GetUndoStack().Any()))
                 {
-                    bool selected = context.GetLastAction() == op;
-                    if (ImGui.Selectable(op.Name, selected))
+                    for(var a = context.GetUndoStack().Count()-1; a >= 0; a--)
                     {
-
+                        context.Undo();
                     }
                 }
-                foreach (var op in context.GetRedoUndoStack())
+                for (var i = 0; i < context.GetUndoStack().Count(); i++)
                 {
-                    ImGui.BeginDisabled();
-
+                    var op = context.GetUndoStack().Reverse().ElementAt(i);
                     bool selected = context.GetLastAction() == op;
-                    if (ImGui.Selectable(op.Name, selected))
+                    if (ImGui.Selectable(op.Name+"##"+i, selected))
                     {
-
+                        for(var a = context.GetUndoStack().Count()-1; a > i; a--)
+                        {
+                            context.Undo();
+                        }
+                    }
+                }
+                for (var i = 0; i < context.GetRedoUndoStack().Count(); i++)
+                {
+                    var op = context.GetRedoUndoStack().ElementAt(i);
+                    ImGui.PushStyleColor(ImGuiCol.Text, new Vector4(1, 1, 1, 0.5f));
```

Comment:
You may want to use `ImGui.GetStyle().Colors[(int)ImGuiCol.TextDisabled]` for the color rather than hardcoding one. Just keeps theming consistent.

Reply:
ok